cmd/core_restart.go

```diff
@@ -24,18 +24,41 @@ Restart the Home Assistant Core instance running on your system`,
 		section := "core"
 		command := "restart"
 
+		url, err := helper.URLHelper(section, command)
+		if err != nil {
+			fmt.Println(err)
+			ExitWithError = true
+			return
+		}
+
+		request := helper.GetRequestTimeout(helper.ContainerOperationTimeout)
+
+		safeMode, err := cmd.Flags().GetBool("safe-mode")
+		if err == nil && safeMode {
+			options := make(map[string]interface{})
+			options["safe_mode"] = safeMode
+			request.SetBody(options)
+		}
+
 		ProgressSpinner.Start()
-		resp, err := helper.GenericJSONPostTimeout(section, command, nil, helper.ContainerOperationTimeout)
```

Comment:
I think it should be possible to continue use `GenericJSONPostTimeout`, it takes an options argument which is set as body.